src/tools.js

```diff
@@ -1,341 +0,0 @@
-const isInvalidHeader = (name, value) => {
-    // NOTE: These are internal Node.js functions, they might stop working in the future!
-    return typeof name !== 'string'
-        || !name
-        || !isHttpToken(name)
-        || value === undefined
-        || isInvalidHeaderChar(value);
-};
-
-module.exports.isInvalidHeader = isInvalidHeader;
-
-const bulletproofDecodeURIComponent = (encodedURIComponent) => {
-    try {
-        return decodeURIComponent(encodedURIComponent);
-    } catch (e) {
-        return encodedURIComponent;
-    }
-};
-
-// Ports returned by `parseUrl` when port is not explicitly specified.
-// Values are based on node docs: https://nodejs.org/api/url.html#url_url_port
-const STANDARD_PORTS_BY_PROTOCOL = {
-    'ftp:': 21,
-    'http:': 80,
-    'https:': 443,
-    'ws:': 80,
-    'wss:': 443,
-};
-
-/**
- * Parses a URL using Node.js' `new URL(url)` and adds the following features:
- *  - `port` is casted to number / null from string
- *  - `path` field is added (pathname + search)
- *  - both username and password is URI-decoded
- *  - `auth` field is added (username + ":" + password, or empty string)
- *
- * Note that compared to the old implementation using `url.parse()`, the new function:
- *  - is unable to distinguish empty password and missing password
- *  - password and username are empty string if not present (or empty)
- *  - we are able to parse IPv6
- *
- * @param url
- * @ignore
- */
-const parseUrl = (url) => {
-    // NOTE: In the past we used url.parse() here, but it can't handle IPv6 and other special URLs,
-    // so we moved to new URL()
-    const urlObj = new URL(url);
-
-    const parsed = {
-        auth: urlObj.username || urlObj.password ? `${urlObj.username}:${urlObj.password}` : '',
-        hash: urlObj.hash,
-        host: urlObj.host,
-        hostname: urlObj.hostname,
-        href: urlObj.href,
-        origin: urlObj.origin,
-        // The username and password might not be correctly URI-encoded, try to make it work anyway
-        username: bulletproofDecodeURIComponent(urlObj.username),
-        password: bulletproofDecodeURIComponent(urlObj.password),
-        pathname: urlObj.pathname,
-        // Path was present on the original UrlObject, it's kept for backwards compatibility
-        path: `${urlObj.pathname}${urlObj.search}`,
-        // Port is turned into a number if available
-        port: urlObj.port ? parseInt(urlObj.port, 10) : null,
-        protocol: urlObj.protocol,
-        scheme: null,
-        search: urlObj.search,
-        searchParams: urlObj.searchParams,
-    };
-
-    // Add scheme field (as some other external tools rely on that)
-    if (parsed.protocol) {
-        const matches = /^([a-z0-9]+):$/i.exec(parsed.protocol);
-        if (matches && matches.length === 2) {
-            // eslint-disable-next-line prefer-destructuring
-            parsed.scheme = matches[1];
-        }
-    }
```

Comment:
I understand why you want to remove it, but I hope it won't cost us large refactorings somewhere in the core codebase.